minetest/minetest
# Add `minetest.settings` to CSM API and allow CSMs to provide `settingtypes.txt`

Add compact, short information about your PR for easier understanding:

Fix #10382
Allow CSMs to use `minetest.settings`.
Allow CSMs to provide `settingtypes.txt`

## To do

This PR is Ready for Review.

- [x] Expose `minetest.settings` to CSM API
- [x] Expose `minetest.setting_get_pos` to CSM API
- [x] Add test to preview CSM
- [x] Allow client mods to provide `settingtypes.txt`

## How to test

Use this simple script inside a CSM and check the console output:
```lua
print(minetest.settings:get_bool("enable_client_modding", false))
```

Open the settings tab and check is there is a `Client Mods` tab.
